### PR TITLE
Refs [TASK][35768] Handle Event News Screen Logic

### PR DESCRIPTION
**Pull request by I-am-Lien**
## Related Tickets
- [#Task 35768: Handle Event News Screen Logic](https://edu-redmine.sun-asterisk.vn/issues/35768)
## WHAT
- Handle Event News Screen Logic
## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://github.com/framgia/coding-standards/tree/master/eng/android |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/sun/excovid19/ui/news/NewsViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            }, {
+                _error.value = it.message.toString()
+            })
+            .addTo(disposables)
+    }
+
+    private fun dataParser(items: List<RSSItem>) {
+        items.forEach {
+            it.description = getLinkImage(it.description)
+        }
```
Comment (datdt-1935) [inline app/src/main/java/com/sun/excovid19/ui/news/NewsViewModel.kt]: return 1 cái List chứ em?

File: app/src/main/java/com/sun/excovid19/utils/ImageExtenstion.kt [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package com.sun.excovid19.utils
+
+import android.widget.ImageView
+import com.bumptech.glide.Glide
+import com.sun.excovid19.R
+
+fun ImageView.loadImage(image: String) {
+    Glide.with(context).load(image)
```
Comment (datdt-1935) [inline app/src/main/java/com/sun/excovid19/utils/ImageExtenstion.kt]:
```suggestion
    Glide.with(context)
        .load(image)
```

File: app/src/main/java/com/sun/excovid19/ui/news/NewsFragment.kt [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    private val newsAdapter = NewsAdapter(::clickItemNews)
+
+    override fun setupViews() {
+    }
+
+    override fun initData() {
+        recycleNews.adapter = newsAdapter
+        binding?.apply {
+            lifecycleOwner = viewLifecycleOwner
+        }
```
Comment (thonc-2135) [inline app/src/main/java/com/sun/excovid19/ui/news/NewsFragment.kt]: có 1 cái dùng apply lm j, bỏ đi

File: app/src/main/java/com/sun/excovid19/ui/news/NewsViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+            .subscribe({
+                dataParser(it)
+                _newItems.value = it
+            }, {
+                _error.value = it.message.toString()
+            })
+            .addTo(disposables)
+    }
+
+    private fun dataParser(items: List<RSSItem>) {
```
Comment (thonc-2135) [inline app/src/main/java/com/sun/excovid19/ui/news/NewsViewModel.kt]: nên đổi tên là parserData

File: app/src/main/java/com/sun/excovid19/ui/news/NewsViewModel.kt [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    private fun getNewsItem() {
+        repository.getNewsItems()
+            .subscribeOn(Schedulers.io())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe({ items ->
+                items.forEach {
+                    it.description = getLinkImage(it.description)
+                }
```
Comment (datdt-1935) [inline app/src/main/java/com/sun/excovid19/ui/news/NewsViewModel.kt]: tách ra map trước đi em

File: app/src/main/java/com/sun/excovid19/ui/news/NewsFragment.kt [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    private val newsAdapter = NewsAdapter(::clickItemNews)
+
+    override fun setupViews() {
+    }
+
+    override fun initData() {
+        recycleNews.adapter = newsAdapter
+        binding?.lifecycleOwner = viewLifecycleOwner
+        viewModel.newItems.observe(viewLifecycleOwner, Observer {
+            (recycleNews.adapter as UpdateDataAdapter<RSSItem>).setData(it)
```
Comment (datdt-1935) [inline app/src/main/java/com/sun/excovid19/ui/news/NewsFragment.kt]: NewsAdapter có phải là subclass của UpdateDataAdapter không em?

Comment (I-am-Lien) [reply app/src/main/java/com/sun/excovid19/ui/news/NewsFragment.kt]: có anh. Em fix rồi

File: app/src/main/java/com/sun/excovid19/ui/news/NewsViewModel.kt [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+            .subscribe({
+                _newItems.value = parserData(it)
+            }, {
+                _error.value = it.message.toString()
+            })
+            .addTo(disposables)
+    }
+
+    private fun parserData(items: List<RSSItem>): List<RSSItem> {
+        items.forEach {
```
Comment (datdt-1935) [inline app/src/main/java/com/sun/excovid19/ui/news/NewsViewModel.kt]: dùng map nhé

File: app/src/main/java/com/sun/excovid19/ui/news/NewsViewModel.kt [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+            .subscribeOn(Schedulers.io())
+            .observeOn(AndroidSchedulers.mainThread())
+            .map {
+                it.map { item ->
+                    RSSItem(
+                        item.title,
+                        getLinkImage(item.description),
+                        item.pubDate,
+                        item.link
+                    )
```
Comment (datdt-1935) [inline app/src/main/java/com/sun/excovid19/ui/news/NewsViewModel.kt]: dùng như này cho gọn nhé
```suggestion
                    item.copy(description = getLinkImage(item.description))
```

**Review by datdt-1935 — APPROVED** [review]

**Pull request merged**